php/helpers.php

```diff
@@ -309,3 +309,136 @@ function block_field_config( $name ) {
+		$block_config['title'] = ucwords( $block_config['name'], '-' );
+		$block_config['title'] = str_replace( '-', ' ', $block_config['title'] );
+	}
+
+	if ( ! isset( $block_config['icon'] ) ) {
+		$block_config['icon'] = 'block_lab';
+	}
+
+	if ( ! isset( $block_config['category'] ) ) {
+		$block_config['category'] = 'common';
+	}
+
+	if ( ! isset( $block_config['excluded'] ) ) {
+		$block_config['excluded'] = array();
+	}
+
+	if ( ! isset( $block_config['keywords'] ) ) {
+		$block_config['keywords'] = array();
+	}
+
+	if ( ! isset( $block_config['fields'] ) ) {
+		$block_config['fields'] = array();
+	}
+
+	add_filter(
+		'block_lab_blocks',
+		function( $blocks ) use ( $block_config ) {
+			if ( ! isset( $block_config['name'] ) ) {
+				return $blocks;
+			}
+
+			if ( isset( $block_config['fields'] ) && is_array( $block_config['fields'] ) ) {
+				foreach ( $block_config['fields'] as $field_name => $field ) {
+					block_lab_add_field( $block_config['name'], $field_name, $field );
+				}
+			}
+
+			$blocks[ "block-lab/{$block_config['name']}" ] = $block_config;
+			return $blocks;
+		}
+	);
+}
+
+/**
+ * Add a field to a block.
+ *
+ * @param string $block_name   The block name (slug), like 'example-block'.
+ * @param string $field_name   The field name (slug), like 'first-name'.
+ * @param array  $field_config {
+ *     An associative array containing the field configuration.
+ *
+ *     @type string $name    The field name.
+ *     @type string $label   The field label.
+ *     @type string $control The field control type. Default: 'text'.
+ *     @type int    $order   The order that the field appears in. Default: 0.
+ *     @type array  $settings {
+ *         An associative array of settings for the field. Each field has a different set of possible settings.
+ *         Check the register_settings method for the field, found in php/blocks/controls/class-{field name}.php.
+ *     }
+ * }
+ */
+function block_lab_add_field( $block_name, $field_name, $field_config = array() ) {
+	$field_config['name'] = str_replace( '_', '-', sanitize_title( $field_name ) );
+
+	if ( ! isset( $field_config['label'] ) ) {
+		$field_config['label'] = ucwords( $field_config['name'], '-' );
+		$field_config['label'] = str_replace( '-', ' ', $field_config['label'] );
+	}
+
+	if ( ! isset( $field_config['control'] ) ) {
+		$field_config['control'] = 'text';
+	}
+
+	$control_class_name  = 'Block_Lab\\Blocks\\Controls\\';
+	$control_class_name .= ucwords( $field_config['control'], '_' );
+	if ( class_exists( $control_class_name ) ) {
+		/**
+		 * An instance of the control, to retrieve the correct type.
+		 *
+		 * @var Control_Abstract $control_class
+		 */
+		$control_class        = new $control_class_name();
+		$field_config['type'] = $control_class->type;
+	}
+
+	if ( ! isset( $field_config['order'] ) ) {
+		$field_config['order'] = 0;
+	}
+
+	if ( ! isset( $field_config['settings'] ) ) {
+		$field_config['settings'] = array();
+	}
+
+	add_filter(
```

Comment:
Like the comment above, what do you think about turning this `add_filter()` call into something like:

```php
block_lab()->loader->add_block( $block_name, $field_config );
```
...or whatever method name.

Reply:
Ah, that won't work very well, as this would need knowledge of existing blocks in order to add field(s) to them.

Would that be a common use case, to add a block via `/wp-admin` > Block Lab > Add New, but then later register a field for it via `block_lab_add_field()`

Reply:
I don't think it would be a very common use case, no.

Reply:
Addressed in https://github.com/getblocklab/block-lab/pull/434/commits/5b98f71fd97a3e6fb6973b89c98f15bbb1993e10.

Reply:
Nice, that looks good. Thanks for pushing it.